### DISCUSSION TITLE
Seen Dev Tools v0.1.4

### DISCUSSION THREAD

**Discussion by abukhadra** [abukhadra/seen#4] [Announcements]
<hr /><em>This discussion was created from the release <a href='https://github.com/abukhadra/seen/releases/tag/v0.1.4'>Seen Dev Tools v0.1.4</a>.</em>

**Comment by PrestigeDevop** [comment]
سلام

it would be cool if you can somehow manged to get generic type conversion to .net clr 
there's was a time where I wanted to create my own parser / interpreter and RTL programming language . 
but later on I wanted to do it just because I could  learn more about    NLP . 
I deprecated my project but I've **seen** -_pun intended_- this project getting updates and GitHub recommended it to me again .. so I'm curious where this project going ? ( I hope it gets some .net's clr  transpiler )

**Reply by abukhadra** [reply]
و عليكم السلام

Apologies for the late reply,
Glad that you are interested,  I too abandoned the project multiple times over the years. My previous attempts focused on syntax and bytecode generation, the issue of course is after lots of effort zero value would be added to anyone interested in programming. this time I am taking a top - bottom approach.  starting with high level code generation.  this guarantees that we stay up to date with what's already out there and allows users to be productive.

Currently the focus is on transpiling code to JS/NodeJS and make use of the existing ecosystem. 

Beside the code, there is a lot of work that needs to be done on the environment and tooling, there is no point in writing your code in Arabic, when all the editors, tools , addresses,  resources  such as images,  links are all named in English. I came up with a few ideas to workaround that too. 

Also there are lots of articles and discussions on forums against the idea of a non-english based language, reason given is that it isolates the community from the rest of the world, I do not believe that this should be the case, and that's why I decided to make it bilingual. and also have future plans to add features to minimize the effort to switch between the 2 languages.

Hopefully details will be document soon on the site.

The very first prototype was written in 2015 transpiling to JVM bytecode. So hopefully when the major features are fully implemented, we can start experimenting with developing a generator that targets the .NET CLR
